src/databricks/labs/ucx/installer/policy.py

```diff
@@ -0,0 +1,131 @@
+        }
+        if conf:
+            for key, value in conf.items():
+                policy_definition[f"spark_conf.{key}"] = self._policy_config(value)
+        if self._ws.config.is_aws:
+            policy_definition["aws_attributes.availability"] = self._policy_config(
+                compute.AwsAvailability.ON_DEMAND.value
+            )
+            if instance_profile:
+                policy_definition["aws_attributes.instance_profile_arn"] = self._policy_config(instance_profile)
+        elif self._ws.config.is_azure:  # pylint: disable=confusing-consecutive-elif
+            policy_definition["azure_attributes.availability"] = self._policy_config(
+                compute.AzureAvailability.ON_DEMAND_AZURE.value
+            )
+        else:
+            policy_definition["gcp_attributes.availability"] = self._policy_config(
+                compute.GcpAvailability.ON_DEMAND_GCP.value
+            )
+        return json.dumps(policy_definition)
+
+    @staticmethod
+    def _extract_external_hive_metastore_conf(cluster_policy):
+        spark_conf_dict = {}
+        instance_profile = None
+        if cluster_policy.get("aws_attributes.instance_profile_arn") is not None:
+            instance_profile = cluster_policy.get("aws_attributes.instance_profile_arn").get("value")
+            logger.info(f"Instance Profile is Set to {instance_profile}")
+        for key in cluster_policy.keys():
+            if (
+                key.startswith("spark_conf.spark.sql.hive.metastore")
+                or key.startswith("spark_conf.spark.hadoop.javax.jdo.option")
+                or key.startswith("spark_conf.spark.databricks.hive.metastore")
+                or key.startswith("spark_conf.spark.hadoop.hive.metastore.glue")
+            ):
+                spark_conf_dict[key[11:]] = cluster_policy[key]["value"]
+        return instance_profile, spark_conf_dict
+
+    def _get_cluster_policies_with_external_hive_metastores(self):
+        for policy in self._ws.cluster_policies.list():
+            def_json = json.loads(policy.definition)
+            glue_node = def_json.get("spark_conf.spark.databricks.hive.metastore.glueCatalog.enabled")
+            if glue_node is not None and glue_node.get("value") == "true":
+                yield policy
+                continue
+            for key in def_json.keys():
+                if key.startswith("spark_conf.spark.sql.hive.metastore"):
+                    yield policy
+                    break
+
+    def update_job_policy(self, states: InstallState, policy_id: str):
+        if not states.jobs:
+            logger.error("No jobs found in states")
+            return
+        for job_id in states.jobs.items():
+            try:
+                job = self._ws.jobs.get(job_id)
+                job_settings = job.settings
+                assert job.job_id is not None
+                assert job_settings is not None
+                assert job_settings.job_clusters is not None
```

Comment:
what if tasks are using existing clusters? we use that with write-protected DBFS - https://github.com/databrickslabs/ucx/blame/main/src/databricks/labs/ucx/install.py#L788-L791 (but that one is not using a policy)